receiver/apachereceiver/testdata/scraper/expected.yaml

```diff
@@ -215,5 +215,5 @@ resourceMetrics:
                   timeUnixNano: "1632495518500962000"
             unit: '{workers}'
         scope:
-          name: otelcol/apachereceiver
+          name: otelcol/apache
```

Comment:
This is interesting. Is this the way we agreed to go? Maybe we should update the template to `otelcol/{{ .Class }}/{{ .Type }}` so we get `otelcol/receiver/apache`. Or `io.opentelemetry.collector.receiver.apache`, which is closer to examples defined in https://github.com/open-telemetry/opentelemetry-specification/blob/f9c26af89dcb2857772e2b75fb1318a521aa05c8/specification/glossary.md#instrumentation-library

cc @open-telemetry/collector-contrib-approvers

Reply:
Yes, this is an unfortunate side-effect of using type more rigorously. Happy to chat about the best way to manage that, we can indeed also change the template.

Reply:
Please look at https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4849307450/jobs/8641252822?pr=21275 as well for context. Right now the type field in metadata is what we are going to attempt to use to check if a component is registered in otelcontribcol components. If the type does not match the typeStr constant folks declare in factory.go, then the test breaks.

Reply:
In terms of the format, the Go SDK uses the module name, which I think would make sense: https://github.com/open-telemetry/opentelemetry-go-contrib/blob/8538a773eb81e44260cb04118ae3002d18abfe32/instrumentation/net/http/otelhttp/config.go#L30. That would make it `github.com/open-telemetry/opentelemetry-collector-contrib/receiver/apachereceiver`.

The [specification examples](https://github.com/open-telemetry/opentelemetry-specification/blob/f9c26af89dcb2857772e2b75fb1318a521aa05c8/specification/overview.md#instrumentation-libraries) for instrumentation libraries use language-specific module formats too, so I think this would make sense.

I'm also fine with the names that @dmitryax suggested, which has the benefit of not being tied to the path of the component in the repository.

Reply:
@evan-bradley That Makes sense. Let's leave `io.opentelemetry.collector.receiver.apache` for java. I wish we had the shortcut to contrib modules as well to have a  `go.opentelemetry.io/collector-contrib/receiver/apachereceiver`

Reply:
+1 to using the package import path for instrumentation scope name.  I believe it is the best option for Go packages.

Regarding the `go.opentelemetry.io` metadata proxy, it takes data from [here](https://github.com/open-telemetry/opentelemetry-go-vanityurls/blob/main/go.opentelemetry.io/vanity.yaml) and adding an entry for `collector-contrib` should be easy.  I believe there are issues with hosting and management that are being worked through ATM, but I would expect the more significant impediment to its use by collector contrib to simply be the accumulated mass of code already built working with modules at the existing import paths that would need to change.

Reply:
I created an issue https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21469. @Aneurysm9 can you please mention/link the hosting and management issues there?